MainWindow.py

```
@@ -1,695 +1,747 @@
+			self.logger.exception("谱面信息更新发生错误: ")
+			# TODO: 修改为root.After
+			self.root.after(0, lambda _:messagebox.showerror("Error", f'发生错误: {e}'))
+			if messagebox.askyesno("无法获取谱面信息更新", f'是否前往网页查看是否存在更新?\n(请比对 SongName.update 中的时间是否比本地文件中的时间更大)'):
+				webbrowser.open("https://raw.githubusercontent.com/Ginsakura/MUSYNCSave/main/musync_data/songname.update")
+		endTime = time.perf_counter_ns()
+		self.logger.info("CheckJsonUpdate Run Time: %f ms"%((endTime - startTime)/1000000))
+
+	def CheckUpdate(self) -> None:
+		def CheckVersion(local:list[int], target:list[int], channel:bool=False)->bool:
+			"""版本号比较"""
+			if (target[0] > local[0]): return True
```

Ruff (0.14.8)

Multiple statements on one line (colon)

(E701)